### ISSUE TITLE
Hex editor shows wrong "File on Disk" when switching games

### ISSUE THREAD

**Issue by kalimag** [TASEmulators/BizHawk#3527]
### Summary
When loading two different roms while the hex editor is open, the "File on Disk" in the hex edtior will show the contents of the previous rom, rather than the current rom. Additionally, if the previous rom was a MAME archive, an error will occur.

### Repro
1. Open the hex editor and keep it open
2. Load any rom A (e.g a N64 rom with an easily identifiable header)
3. Load a different rom B
4. Switch to the "File on Disk" domain in the hex editor
5. "File on Disk" will display contents of rom A

Alternatively, load a MAME archive at step 2 to reproduce the exception

### Output
If the first rom was a MAME archive:
```
HawkFile: Can't call GetStream() before you've successfully bound something!
   at BizHawk.Common.HawkFile.GetStream() in F:\Emulation\BizHawk\src\BizHawk.Common\HawkFile\HawkFile.cs:line 267
   at BizHawk.Client.EmuHawk.HexEditor.GetRomBytes() in F:\Emulation\BizHawk\src\BizHawk.Client.EmuHawk\tools\HexEditor\HexEditor.cs:line 446
   at BizHawk.Client.EmuHawk.HexEditor.Restart() in F:\Emulation\BizHawk\src\BizHawk.Client.EmuHawk\tools\HexEditor\HexEditor.cs:line 210
   at BizHawk.Client.EmuHawk.ToolManager.Restart(Config config, IEmulator emulator, IGameInfo game) in F:\Emulation\BizHawk\src\BizHawk.Client.EmuHawk\tools\ToolManager.cs:line 521
   at BizHawk.Client.EmuHawk.MainForm.LoadRomInternal(String path, LoadRomArgs args, Boolean& failureIsFromAskSave) in F:\Emulation\BizHawk\src\BizHawk.Client.EmuHawk\MainForm.cs:line 3934
   at BizHawk.Client.EmuHawk.MainForm.LoadRom(String path, LoadRomArgs args, Boolean& failureIsFromAskSave) in F:\Emulation\BizHawk\src\BizHawk.Client.EmuHawk\MainForm.cs:line 3714
   at BizHawk.Client.EmuHawk.MainForm.LoadRom(String path, LoadRomArgs args) in F:\Emulation\BizHawk\src\BizHawk.Client.EmuHawk\MainForm.cs:line 3710
   at BizHawk.Client.EmuHawk.MainForm.OpenRom() in F:\Emulation\BizHawk\src\BizHawk.Client.EmuHawk\MainForm.cs:line 2427
   at BizHawk.Client.EmuHawk.MainForm.OpenRomMenuItem_Click(Object sender, EventArgs e) in F:\Emulation\BizHawk\src\BizHawk.Client.EmuHawk\MainForm.Events.cs:line 296
```

### Host env.
- BizHawk 2.3 (earliest tested) up to 2.9 dev

### Cause

The issue is that `HexEditor.GetRomBytes()` (called from `HexEditor.Restart()`) loads the rom from `MainForm.CurrentlyOpenRomArgs`, but `MainForm.LoadRomInternal` calls `Tools.Restart()` before updating `CurrentlyOpenRomArgs`:

https://github.com/TASEmulators/BizHawk/blob/1df6ce4e38946564765dab9227ed093b5f33812f/src/BizHawk.Client.EmuHawk/tools/HexEditor/HexEditor.cs#L425-L433

https://github.com/TASEmulators/BizHawk/blob/15056f11f2ac1f1b186e9144a259cc35d9bd633e/src/BizHawk.Client.EmuHawk/MainForm.cs#L3934-L3950

**Comment by kalimag** [comment]
In general, I think calling `Tools.Restart` before updating `CurrentlyOpenRom`/`CurrentlyOpenRomArgs` is unexpected and undesirable.

Other places where those properties are read inside `Restart`:

https://github.com/TASEmulators/BizHawk/blob/1df6ce4e38946564765dab9227ed093b5f33812f/src/BizHawk.Client.EmuHawk/tools/HexEditor/HexEditor.cs#L227-L232

https://github.com/TASEmulators/BizHawk/blob/ec6fe5fcf18b48f85b81ed80b00220392290537b/src/BizHawk.Client.EmuHawk/tools/BasicBot/BasicBot.cs#L288-L292

https://github.com/TASEmulators/BizHawk/blob/3f4aee0154b05fcc3da36695d7ecd43f4a3859f0/src/BizHawk.Client.EmuHawk/tools/MultiDiskBundler/MultiDiskBundler.cs#L56-L90

Could also negatively affect external tools.

**Comment by YoshiRulz** [comment]
Context: these are `((this as ToolFormBase).MainForm as IMainFormForTools).*` and so have found their way into ApiHawk. But whether they were ever meant to be isn't really relevant when the current behaviour also breaks first-party tools.

**Comment by kalimag** [comment]
Resolved by #3691
